composer/trainer/mosaic_fsdp_utils.py

```diff
@@ -1033,3 +1033,48 @@ def _shard_orig_param_state(
             new_optim_state[state_name] = value
         torch.cuda.synchronize()
         return new_optim_state
+
+
+    from torch.distributed.fsdp import _state_dict_utils
+    from torch.distributed.fsdp._common_utils import _FSDPState
+    from torch.distributed.fsdp.api import FullStateDictConfig, StateDictType
+
+    @no_type_check
+    def _full_pre_state_dict_hook(
```

Comment:
can you describe the patch? https://github.com/mosaicml/llm-foundry-private/pull/122#discussion_r1504714070